dffml/df/base.py

```diff
@@ -413,6 +387,52 @@ async def run(
     return wrap
 
 
+def createDefinition(name, name_list, param_annotation):
+    if param_annotation in primitive_types:
+        return Definition(
+            name=".".join(name_list),
+            primitive=primitive_convert.get(
+                param_annotation, param_annotation.__name__
+            ),
+        )
+    elif (
+        get_origin(param_annotation) is list
+        or get_origin(param_annotation) is dict
+    ):
+        # If the annotation are of the form List[MyDataClass] or Dict[Any, MyDataClass]
+        if get_origin(param_annotation) is list:
+            primitive = "array"
+            innerclass = list(get_args(param_annotation))[0]
+        else:
+            primitive = "map"
+            innerclass = list(get_args(param_annotation))[1]
+
+        if issubclass(type(innerclass), type(Any)):
+            # if the innerclass is Any type ie. Dict[str, Any]
+            return None
```

Comment:
If we see this then we should have a body similar to the first case, `if param_annotation in primitive_types:` where `param_annotation` is now `dict`, we should do the same for if we see `List[Any]`

But why have a similar block of code, when we can have the same block of code!

Let's just recuse into the function so that it triggers the first if block.

```python
        if issubclass(type(innerclass), type(Any)):
            return create_defintition(name, get_origin(param_annotation))

Reply:
yeah, so.... let me explain why we have different block of code for `List[Data]` or `Dict[str, Data]`.
```python
>>> import typing
>>> from typing import List, Dict
>>> import inspect
>>> from typing import List, Dict, NamedTuple
>>> 
>>> class Data(NamedTuple):
...     name: str
...     age: int
... 
>>> def some(a: int) -> List[Data]:
...     pass
... 
>>> def some2(a: int) -> Dict[str, Data]:
...     pass
... 
>>> return_type = inspect.signature(some).return_annotation
>>> print(return_type)
typing.List[__main__.Data]
>>> isinstance(return_type, list)
False
>>> isinstance(return_type, typing.List)
False
>>> isinstance(return_type, typing._GenericAlias)
True
>>> type(return_type)
<class 'typing._GenericAlias'>
>>> 
>>> return_type = inspect.signature(some2).return_annotation
>>> type(return_type)
<class 'typing._GenericAlias'>
>>> isinstance(return_type, dict)
False
>>> isinstance(return_type, typing.Dict)
False
>>> isinstance(return_type, typing._GenericAlias)
True
```
So actually the annotation is `typing._GenericAlias` but when we do `get_origin(return_type)` then it matches to `list` or `dict` types.
And also when we try to do `get_origin` on `list` or `dict` types it returns `None`